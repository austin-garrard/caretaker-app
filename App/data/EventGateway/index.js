/*
  Makes API calls to fetch events.

  However, until there is an API to call, it returns canned data.
*/

export default class EventGateway {

    static allEvents = null;
    static subscribers = [];

    static getAll() {
        if(this.allEvents === null) {
            //api call to get all events
            this.allEvents = [
                {id: 1, startDate: 1498932000000, endDate: 1498935600000, date: 'July 1', time: '13:00-14:00', location: '6818 Austin Center Blvd, Austin, TX 78731', name: 'Chemotherapy', volunteerId: 'jack@coolwebsite.com', role: 'driver', description: ''},
                {id: 2, startDate: 1499097600000, endDate: 1499101200000, date: 'July 3rd', time: '11:00-12:00', location: '6818 Austin Center Blvd, Austin, TX 78731', name: 'Toxicity Check', volunteerId: 'sarah@emailprovider.com', role: 'driver', description: 'Quick check'},
                {id: 3, startDate: 1499261400000, endDate: 1499265000000, date: 'July 5th', time: '08:30-09:30', location: '1120 S Lamar Blvd, Austin, TX 78704', name: 'Checkin', volunteerId: 'TBD', role: 'active friend', description: ''},
                {id: 4, startDate: 1499403600000, date: 'July 7th', time: 'All Day', location: '1100 Congress Ave, Austin, TX 78701', name: 'Weekend', volunteerId: 'sarah@emailprovider.com', role: 'active friend', description: 'Need to keep busy'},
                {id: 5, startDate: 1499613300000, endDate: 1499620500000, date: 'July 9th', time: '10:15-12:15', location: '6818 Austin Center Blvd, Austin, TX 78731', name: 'Chemotherapy', volunteerId: 'TBD', role: 'driver', description: ''},
                {id: 6, startDate: 1499699700000, endDate: 1499706900000, date: 'July 10th', time: '10:15-12:15', location: '6818 Austin Center Blvd, Austin, TX 78731', name: 'Chemotherapy', volunteerId: 'austin@yeehaw.com', role: 'driver', description: ''}
            ]
        }
        return this.allEvents;
    }

    static pickUpEvent(eventId, volunteerId) {
        let event = this.allEvents.find((event) => event.id === eventId)
        event.volunteerId = volunteerId;
        //api call to update event by id
        this.notifySubscribers();
    }

    static dropEvent(eventId) {
        let event = this.allEvents.find((event) => event.id === eventId)
        event.volunteerId = 'TBD';
        //api call to update event by id
        this.notifySubscribers();
    }

    static createNewEvent(data) {
      //api call to create event
      const result = {id: 7, startDate: 1499699700000, endDate: 1499706900000, location: '6818 Austin Center Blvd, Austin, TX 78731', name: 'New Event 1', volunteerId: 'austin@yeehaw.com', role: 'driver', description: ''}
      this.allEvents.push(result);
      alert('created event with data ' + JSON.stringify(data));
      this.notifySubscribers();
    }

    static subscribeToUpdates(callback) {
        this.subscribers.push(callback);
    }

    static notifySubscribers() {
        this.subscribers.forEach((subscriber) => subscriber());
    }

}
