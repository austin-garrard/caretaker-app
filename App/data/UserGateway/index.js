export class Permissions {
    static FOCUS = 'focus';
    static ADMIN = 'admin';
    static HELPER = 'helper';
}

export default class UserGateway {

    static currentUser = null;
    static allUsers = null;

    static getAll() {
        if(this.allUsers === null) {
            //api call goes here
            this.allUsers = [
                {name: 'Sarah', identifier: 'sarah@emailprovider.com', permission: Permissions.FOCUS, roles: [], phone:'123-456-7890'},
                {name: 'Sarah', identifier: 's', permission: Permissions.FOCUS, roles: [], phone:'123-456-7890'},
                {name: 'Caroline', identifier: 'caroline@woahdude.com', permission: Permissions.ADMIN, roles: ['driver', 'coordinator'], phone:'123-456-7890'},
                {name: 'Jack', identifier: 'jack@coolwebsite.com', permission: Permissions.HELPER, roles: ['active friend']},
                {name: 'Austin', identifier: 'austin@yeehaw.com', permission: Permissions.HELPER, roles: [], phone:'123-456-7890'}
            ]
        }
        return this.allUsers;
    }

    static getName(identifier) {
        const user = this.allUsers.find((user) => user.identifier === identifier)
        return user ? user.name : 'TBD';
    }

    static getRoles() {
        return this.currentUser.roles
    }

    static signIn(email) {
        //make call to google oauth and store the results
        let results = {
            email: email
        };
        let userFromBackend = this.getAll().find((user) => user.identifier === results.email);
        if(typeof userFromBackend !== 'undefined') {
            this.currentUser = userFromBackend;
            return true;
        }
        return false;
    }

    static isSelf(identifier) {
        return this.currentUser.identifier === identifier;
    }

    static isAdmin() {
        return this.currentUser.permission === Permissions.ADMIN
            || this.currentUser.permission === Permissions.FOCUS;
    }

    static getCurrentPermissions() {
        return this.currentUser.permission;
    }

    static inviteUser(email) {
        //api call to invite the user
        alert('invited ' + email + '!');
    }
}
