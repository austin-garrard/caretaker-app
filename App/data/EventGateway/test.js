import EventGateway from './index.js';

describe('Event Gateway', () => {

    it('should get all the events', () => {
        expect(EventGateway.getAll()).toEqual([
            {id: 1, startDate: 1498932000000, endDate: 1498935600000, date: 'July 1', time: '13:00-14:00', location: '6818 Austin Center Blvd, Austin, TX 78731', name: 'Chemotherapy', volunteerId: 'jack@coolwebsite.com', role: 'driver', description: ''},
            {id: 2, startDate: 1499097600000, endDate: 1499101200000, date: 'July 3rd', time: '11:00-12:00', location: '6818 Austin Center Blvd, Austin, TX 78731', name: 'Toxicity Check', volunteerId: 'sarah@emailprovider.com', role: 'driver', description: 'Quick check'},
            {id: 3, startDate: 1499261400000, endDate: 1499265000000, date: 'July 5th', time: '08:30-09:30', location: '1120 S Lamar Blvd, Austin, TX 78704', name: 'Checkin', volunteerId: 'TBD', role: 'active friend', description: ''},
            {id: 4, startDate: 1499403600000, date: 'July 7th', time: 'All Day', location: '1100 Congress Ave, Austin, TX 78701', name: 'Weekend', volunteerId: 'sarah@emailprovider.com', role: 'active friend', description: 'Need to keep busy'},
            {id: 5, startDate: 1499613300000, endDate: 1499620500000, date: 'July 9th', time: '10:15-12:15', location: '6818 Austin Center Blvd, Austin, TX 78731', name: 'Chemotherapy', volunteerId: 'TBD', role: 'driver', description: ''},
            {id: 6, startDate: 1499699700000, endDate: 1499706900000, date: 'July 10th', time: '10:15-12:15', location: '6818 Austin Center Blvd, Austin, TX 78731', name: 'Chemotherapy', volunteerId: 'austin@yeehaw.com', role: 'driver', description: ''}
        ]);
    });

    it('should notify subscribers when an event is picked up', () => {
        const subscriber = jest.fn();
        EventGateway.subscribeToUpdates(subscriber);

        EventGateway.pickUpEvent(1, 'austin@yeehaw.com');

        expect(subscriber).toHaveBeenCalled();
    })

    it('should notify subscribers when an event is dropped', () => {
        const subscriber = jest.fn();
        EventGateway.subscribeToUpdates(subscriber);

        EventGateway.dropEvent(1);

        expect(subscriber).toHaveBeenCalled();
    })

    it('should notify subscribers when an event is created', () => {
        const subscriber = jest.fn();
        EventGateway.subscribeToUpdates(subscriber);

        EventGateway.createNewEvent({some: 'data'});

        expect(subscriber).toHaveBeenCalled();
    })
})
