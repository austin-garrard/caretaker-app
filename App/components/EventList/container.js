import React, {Component} from 'react';
import EventGateway from '../../data/EventGateway';
import UserGateway from '../../data/UserGateway';

export var ContainerFor = (EventList) => class extends Component {
    constructor() {
        super();
        this.state = {
            events: [],
            visibleModal: false,
            selectedEvent: null,
        };
        EventGateway.subscribeToUpdates(this.updateEvents);
    }

    updateEvents = () => {
        let events = EventGateway.getAll();
        events.forEach((event) => event.volunteer = UserGateway.getName(event.volunteerId));
        let newState = Object.assign({}, this.state);
        newState.events = events;
        this.setState(newState);
    }

    componentDidMount() {
        this.updateEvents();
    }

    _onPressItem = (eventId) => {
        let newState = this.state;
        newState.visibleModal = true;
        newState.selectedEvent = newState.events.find((event) => event.id === eventId)
        this.setState(newState);
    };

    _onCloseModal = () => {
        let newState = this.state;
        newState.visibleModal = false;
        this.setState(newState);
    }

    _onCreateEvent = (data) => {
      alert('hi');
      EventGateway.createNewEvent(data);
    }

    render() {
        return <EventList
            events={this.state.events}
            visibleModal={this.state.visibleModal}
            selectedEvent={this.state.selectedEvent}
            onPressItem={this._onPressItem}
            onCloseModal={this._onCloseModal}
            onCreateEvent={this._onCreateEvent}
        />;
    }
}
