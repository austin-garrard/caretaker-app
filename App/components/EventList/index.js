import React, {Component, PureComponent} from 'react';
import {
    FlatList,
    StyleSheet,
    Text,
    TouchableOpacity,
    View
} from 'react-native';
import { ActionButton } from 'react-native-material-ui';
import {ContainerFor} from './container.js';
import EventModal from '../shared/EventModal/';

class MyListItem extends PureComponent {
    _onPress = () => {
        this.props.onPressItem(this.props.id);
    };

    render() {
        return (
            <TouchableOpacity onPress={this._onPress}>
                <View style={[styles.button, styles.eventBox]}>
                  <View style={styles.eventDateWrap}>
                    <Text style={styles.eventDateDay}>{this.props.startDate.getDate()}</Text>
                    <Text style={styles.eventDateDayName}>{['Sun', 'Mon', 'Tue', 'Wed', 'Thu', 'Fri', 'Sat'][this.props.startDate.getDay()]}</Text>
                  </View>
                  <View style={styles.eventDetailsWrap}>
                    <Text style={styles.eventName}>{this.props.name}</Text>
                    <Text style={styles.eventVolunteer}>{this.props.volunteer}</Text>
                  </View>
                </View>
            </TouchableOpacity>
        );
    }
}

class EventList extends Component {

    _renderItem = ({item}) => (
        <MyListItem
            id={item.id}
            onPressItem={this.props.onPressItem}
            title={item.title}
            startDate={new Date(item.startDate)}
            endDate={new Date(item.startDate)}
            date={item.date}
            name={item.name}
            volunteer={item.volunteer}
        />
    );

    _onCreateEvent = () => {
      this.props.onCreateEvent({some: 'data'});
    }

    render() {
        return (
            <View style={styles.container}>
                <EventModal
                    event={this.props.selectedEvent}
                    visible={this.props.visibleModal}
                    onClose={this.props.onCloseModal}
                />
                <FlatList
                   style={styles.eventList}
                   data={this.props.events}
                   dataWhichCouldChange={this.props.events.map((event) => event.volunteer)}
                   renderItem={this._renderItem}
                   keyExtractor={(item) => item.id}
                />
                <ActionButton onPress={this._onCreateEvent} />
            </View>
        );
    }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#F5FCFF'
  },
  title: {
    fontSize: 24,
    textAlign: 'center',
    margin: 15,
  },
  item: {
    fontSize: 20,
    textAlign: 'left',
    margin: 13,
  },
  modalContent: {
    margin: 13,
  },
  modalTitle: {
    fontSize: 24,
    textAlign: 'center',
    margin: 15
  },
  modalInfo: {
    fontSize: 20,
    textAlign: 'left',
    margin: 10
  },
  eventBox: {
    flex: 1,
    flexDirection: 'row',
    marginTop: 4,
    marginBottom: 0,
    marginLeft: 8,
    marginRight: 8
  },
  eventDateWrap: {
    width: 56,
    padding: 4
  },
  eventDateDay: {
    fontSize: 24
  },
  eventDetailsWrap: {
    flex: 1,
    borderWidth: 1,
    borderStyle: 'solid',
    borderColor: '#FF0000',
    borderRadius: 2,
    padding: 4
  },
  eventName: {
    fontWeight: 'bold'
  },
  eventVolunteer: {},
  eventList: {
    backgroundColor: '#FFF'
  }
});

export default ContainerFor(EventList);
