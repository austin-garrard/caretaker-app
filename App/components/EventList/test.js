import React, {Component} from 'react';
import renderer from 'react-test-renderer';
import EventList from './index.js';
import UserGateway from '../../data/UserGateway';
import EventGateway from '../../data/EventGateway';

describe('EventList', () => {
    beforeEach(() => {
        UserGateway.signIn('sarah@emailprovider.com'); //todo: mock/stub gateways
    });

    it('should render correctly', () => {
        const list = renderer.create(
            <EventList />
        )

        expect(list).toMatchSnapshot();
    });

    it('should subscribe to updates from the event gateway', () => {
        const spy = jest.spyOn(EventGateway, 'subscribeToUpdates');

        const list = renderer
            .create(<EventList />)
            .getInstance();

        expect(spy).toHaveBeenCalledWith(list.updateEvents);
    });

    it('should get all the events when first rendered', () => {
        const spy = jest.spyOn(EventGateway, 'getAll');

        const list = renderer
            .create(<EventList />)
            .getInstance();

        expect(spy).toHaveBeenCalled();
    });

    it('it should get each volunteer name from the user gateway', () => {
        const spy = jest.spyOn(UserGateway, 'getName');
        const events = EventGateway.getAll(); //todo: stub this out

        const list = renderer
            .create(<EventList />)
            .getInstance();

        events.forEach((event) => expect(spy).toHaveBeenCalledWith(event.volunteerId));
    });

    describe('_onPressItem', () => {
        it('should make the modal visible', () => {
            const list = renderer
                .create(<EventList />)
                .getInstance();

            list._onPressItem(1);

            expect(list.state.visibleModal).toBe(true);
        });

        it('should set the selectedEvent', () => {
            const events = EventGateway.getAll(); //todo: stub this out
            const list = renderer
                .create(<EventList />)
                .getInstance();

            list._onPressItem(1);

            expect(list.state.selectedEvent).toBe(events[0]);
        });
    });

    describe('_onCloseModal', () => {
        it('should make the modal not visible', () => {
            const list = renderer
                .create(<EventList />)
                .getInstance();

            list._onCloseModal();

            expect(list.state.visibleModal).toBe(false);
        });
    });

    describe('_onCreateEvent', () => {
      it('should create an event through the gateway', () => {
        const spy = jest.spyOn(EventGateway, 'createNewEvent');
        const list = renderer
            .create(<EventList />)
            .getInstance();

          list._onCreateEvent({some: 'data'});

          expect(spy).toHaveBeenCalledWith({some: 'data'});
      })
    })
});
