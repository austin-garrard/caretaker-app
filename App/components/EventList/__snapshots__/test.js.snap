// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventList should render correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#F5FCFF",
      "flex": 1,
    }
  }
>
  <RCTScrollView
    data={
      Array [
        Object {
          "date": "July 1",
          "description": "",
          "endDate": 1498935600000,
          "id": 1,
          "location": "6818 Austin Center Blvd, Austin, TX 78731",
          "name": "Chemotherapy",
          "role": "driver",
          "startDate": 1498932000000,
          "time": "13:00-14:00",
          "volunteer": "Jack",
          "volunteerId": "jack@coolwebsite.com",
        },
        Object {
          "date": "July 3rd",
          "description": "Quick check",
          "endDate": 1499101200000,
          "id": 2,
          "location": "6818 Austin Center Blvd, Austin, TX 78731",
          "name": "Toxicity Check",
          "role": "driver",
          "startDate": 1499097600000,
          "time": "11:00-12:00",
          "volunteer": "Sarah",
          "volunteerId": "sarah@emailprovider.com",
        },
        Object {
          "date": "July 5th",
          "description": "",
          "endDate": 1499265000000,
          "id": 3,
          "location": "1120 S Lamar Blvd, Austin, TX 78704",
          "name": "Checkin",
          "role": "active friend",
          "startDate": 1499261400000,
          "time": "08:30-09:30",
          "volunteer": "TBD",
          "volunteerId": "TBD",
        },
        Object {
          "date": "July 7th",
          "description": "Need to keep busy",
          "id": 4,
          "location": "1100 Congress Ave, Austin, TX 78701",
          "name": "Weekend",
          "role": "active friend",
          "startDate": 1499403600000,
          "time": "All Day",
          "volunteer": "Sarah",
          "volunteerId": "sarah@emailprovider.com",
        },
        Object {
          "date": "July 9th",
          "description": "",
          "endDate": 1499620500000,
          "id": 5,
          "location": "6818 Austin Center Blvd, Austin, TX 78731",
          "name": "Chemotherapy",
          "role": "driver",
          "startDate": 1499613300000,
          "time": "10:15-12:15",
          "volunteer": "TBD",
          "volunteerId": "TBD",
        },
        Object {
          "date": "July 10th",
          "description": "",
          "endDate": 1499706900000,
          "id": 6,
          "location": "6818 Austin Center Blvd, Austin, TX 78731",
          "name": "Chemotherapy",
          "role": "driver",
          "startDate": 1499699700000,
          "time": "10:15-12:15",
          "volunteer": "Austin",
          "volunteerId": "austin@yeehaw.com",
        },
      ]
    }
    dataWhichCouldChange={
      Array [
        "Jack",
        "Sarah",
        "TBD",
        "Sarah",
        "TBD",
        "Austin",
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReached={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onViewableItemsChanged={undefined}
    removeClippedSubviews={true}
    renderItem={[Function]}
    renderScrollComponent={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    style={
      Object {
        "backgroundColor": "#FFF",
      }
    }
    updateCellsBatchingPeriod={50}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hitSlop={undefined}
          isTVSelectable={true}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <View
            style={
              Array [
                undefined,
                Object {
                  "flex": 1,
                  "flexDirection": "row",
                  "marginBottom": 0,
                  "marginLeft": 8,
                  "marginRight": 8,
                  "marginTop": 4,
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "padding": 4,
                  "width": 56,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "fontSize": 24,
                  }
                }
              >
                1
              </Text>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={undefined}
              >
                Sat
              </Text>
            </View>
            <View
              style={
                Object {
                  "borderColor": "#FF0000",
                  "borderRadius": 2,
                  "borderStyle": "solid",
                  "borderWidth": 1,
                  "flex": 1,
                  "padding": 4,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "fontWeight": "bold",
                  }
                }
              >
                Chemotherapy
              </Text>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={Object {}}
              >
                Jack
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hitSlop={undefined}
          isTVSelectable={true}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <View
            style={
              Array [
                undefined,
                Object {
                  "flex": 1,
                  "flexDirection": "row",
                  "marginBottom": 0,
                  "marginLeft": 8,
                  "marginRight": 8,
                  "marginTop": 4,
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "padding": 4,
                  "width": 56,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "fontSize": 24,
                  }
                }
              >
                3
              </Text>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={undefined}
              >
                Mon
              </Text>
            </View>
            <View
              style={
                Object {
                  "borderColor": "#FF0000",
                  "borderRadius": 2,
                  "borderStyle": "solid",
                  "borderWidth": 1,
                  "flex": 1,
                  "padding": 4,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "fontWeight": "bold",
                  }
                }
              >
                Toxicity Check
              </Text>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={Object {}}
              >
                Sarah
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hitSlop={undefined}
          isTVSelectable={true}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <View
            style={
              Array [
                undefined,
                Object {
                  "flex": 1,
                  "flexDirection": "row",
                  "marginBottom": 0,
                  "marginLeft": 8,
                  "marginRight": 8,
                  "marginTop": 4,
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "padding": 4,
                  "width": 56,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "fontSize": 24,
                  }
                }
              >
                5
              </Text>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={undefined}
              >
                Wed
              </Text>
            </View>
            <View
              style={
                Object {
                  "borderColor": "#FF0000",
                  "borderRadius": 2,
                  "borderStyle": "solid",
                  "borderWidth": 1,
                  "flex": 1,
                  "padding": 4,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "fontWeight": "bold",
                  }
                }
              >
                Checkin
              </Text>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={Object {}}
              >
                TBD
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hitSlop={undefined}
          isTVSelectable={true}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <View
            style={
              Array [
                undefined,
                Object {
                  "flex": 1,
                  "flexDirection": "row",
                  "marginBottom": 0,
                  "marginLeft": 8,
                  "marginRight": 8,
                  "marginTop": 4,
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "padding": 4,
                  "width": 56,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "fontSize": 24,
                  }
                }
              >
                7
              </Text>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={undefined}
              >
                Fri
              </Text>
            </View>
            <View
              style={
                Object {
                  "borderColor": "#FF0000",
                  "borderRadius": 2,
                  "borderStyle": "solid",
                  "borderWidth": 1,
                  "flex": 1,
                  "padding": 4,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "fontWeight": "bold",
                  }
                }
              >
                Weekend
              </Text>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={Object {}}
              >
                Sarah
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hitSlop={undefined}
          isTVSelectable={true}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <View
            style={
              Array [
                undefined,
                Object {
                  "flex": 1,
                  "flexDirection": "row",
                  "marginBottom": 0,
                  "marginLeft": 8,
                  "marginRight": 8,
                  "marginTop": 4,
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "padding": 4,
                  "width": 56,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "fontSize": 24,
                  }
                }
              >
                9
              </Text>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={undefined}
              >
                Sun
              </Text>
            </View>
            <View
              style={
                Object {
                  "borderColor": "#FF0000",
                  "borderRadius": 2,
                  "borderStyle": "solid",
                  "borderWidth": 1,
                  "flex": 1,
                  "padding": 4,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "fontWeight": "bold",
                  }
                }
              >
                Chemotherapy
              </Text>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={Object {}}
              >
                TBD
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hitSlop={undefined}
          isTVSelectable={true}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <View
            style={
              Array [
                undefined,
                Object {
                  "flex": 1,
                  "flexDirection": "row",
                  "marginBottom": 0,
                  "marginLeft": 8,
                  "marginRight": 8,
                  "marginTop": 4,
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "padding": 4,
                  "width": 56,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "fontSize": 24,
                  }
                }
              >
                10
              </Text>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={undefined}
              >
                Mon
              </Text>
            </View>
            <View
              style={
                Object {
                  "borderColor": "#FF0000",
                  "borderRadius": 2,
                  "borderStyle": "solid",
                  "borderWidth": 1,
                  "flex": 1,
                  "padding": 4,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "fontWeight": "bold",
                  }
                }
              >
                Chemotherapy
              </Text>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={Object {}}
              >
                Austin
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "height": 0,
          }
        }
      />
    </View>
  </RCTScrollView>
  <ActionButton
    onPress={[Function]}
  />
</View>
`;
